ports/libsimpleservo/capi/src/lib.rs

```diff
@@ -212,13 +264,19 @@ pub extern "C" fn servo_version() -> *const c_char {
 }
 
 #[cfg(target_os = "windows")]
-fn init_logger() {
+fn init_logger(_opts: &CInitOptions) {
     use log::LevelFilter;
     use std::sync::Once;
     use vslogger::VSLogger;
 
     static LOGGER: VSLogger = VSLogger;
     static LOGGER_INIT: Once = Once::new();
+
+    if _opts.vslogger_mod_size > 0 {
+        LOG_MODULE_FILTERS.lock().unwrap().vslogger_filter =
+            LOGGER.add_module_to_filter(_opts.vslogger_mod_list, _opts.vslogger_mod_size);
```

Comment:
Instead of calling a method here, let's doing the conversion inline. If we accept a `modules: &[*const c_char]` argument, then we can write:
```rust
*LOG_MODULE_FILTERS.lock().unwrap() = modules
    .iter()
    .map(|module| unsafe {
            CStr::from_ptr(module).to_string_lossy().into_owned()
    })
    .collect::<Vec<_>>();
```

Reply:
Implemented.

Reply:
While running the hololens application, I get an access  violation error immediately after stepping over:
```
*LOG_MODULE_FILTERS.lock().unwrap() = _modules
        .iter()
        .map(|_modules| unsafe { CStr::from_ptr(*_modules).to_string_lossy().into_owned() })
        .collect::<Vec<_>>();
```
in the function
```
#[cfg(target_os = "windows")]
fn init_logger(_modules: &[*const c_char]) {
    use log::LevelFilter;
    use std::sync::Once;
    use vslogger::VSLogger;
	use crate::vslogger::LOG_MODULE_FILTERS;

    static LOGGER: VSLogger = VSLogger;
    static LOGGER_INIT: Once = Once::new();

    *LOG_MODULE_FILTERS.lock().unwrap() = _modules
        .iter()
        .map(|_modules| unsafe { CStr::from_ptr(*_modules).to_string_lossy().into_owned() })
        .collect::<Vec<_>>();

    LOGGER_INIT.call_once(|| {
        log::set_logger(&LOGGER)
            .map(|_| log::set_max_level(LevelFilter::Debug))
            .unwrap();
    });
}
```
I do have a filter module setup in HoloLens, but the same exception occurs if I don't.
What is odd is the length for _modules.

data_ptr	0x00007ff62cb5c5cc {ServoApp.exe!`string'} "servo" {0x0000006f76726573 <Error reading characters of string.>}	char * *	
length	140695288792536	unsigned __int64

Reply:
BTW, here is the implementation for:
 ```
lazy_static! {
    pub static ref LOG_MODULE_FILTERS: Arc<Mutex<Vec<String>>> = Arc::new(Mutex::new(vec![]));
}
```

Reply:
It's not clear to me from this snippet where the problem lies. Perhaps verify a couple things - is init_logger only called once? How many entries does `_modules.len()` report when init_logger() is called? What is the value of opts.vslogger_mod_size in the caller?

Reply:
Problem solved.  Should have realized that the problem stemmed from servo.cpp in Hololens when I saw the _modules length of 140695288792536.

I previously had *mods[64] = {....}.  It should have been *mods[] = { ... }.

Also, the number of modules calcuation needed to change to:
sizeof(mods)/sizeof(mods[0]) - 1;